src/mapblock.cpp

```diff
@@ -592,9 +634,17 @@ void MapBlock::deSerialize(std::istream &in_compressed, u8 version, bool disk)
 			m_node_timers.deSerialize(is, version);
 		}
 
-		u16 dummy;
-		m_is_air = nimap.size() == 1 && nimap.getId("air", dummy);
-		m_is_air_expired = false;
+		if (nimap.size() == 1) {
+			u16 dummy;
+			if (nimap.getId("air", dummy)) {
+				reallocate(1, data[0]);
+				m_is_mono_block = true;
```

Comment:
this is unsound (param1, param2)

Reply:
Does it matter for air-only?

Reply:
Aside from arbitrary param2-data, sure, for light data?

Reply:
OK... I'll just do the exact check in this case, might impact de-serialization speed, but I'll test that.